dxp/liferay_global_event_handler.md

```diff
@@ -0,0 +1,213 @@
+# Liferay’s Global Event Handler
+
+Liferay’s event handler helps the developers for creating/listening/firing/removing custom events that can be listened in whole DXP which is rely on the YUI3 event target implementation. See the implementation [here](https://github.com/liferay/yui3/blob/master/src/event-custom/js/event-target.js#L120).
+
+### First Considerations
+
+First, just consider using the global event handler if necessary. Try to use the default event listener/[callback pattern(if React)](https://reactjs.org/docs/lifting-state-up.html#lifting-state-up) when working. Just use the global event handler when trying to communicate between widgets or not accessible contexts(e.g iframes).
```

Comment:
Typos: Missing space before `(` in two places.

> First, just consider using the global event handler if necessary.

It's not clear to me what this sentence means:

- It _sounds_ like using `document.addEventListener` (something which should be kept to a minimum anyway); or:
- You mean something else by "the global event handler", like "don't use the Liferay global event without first trying to use a more local solution.

Based on the context, I think you mean the second one, right?

> the default event listener

It's also not clear to me what you're referring to there.

I think this whole paragraph might be clearer overall as something like:

> In many situations you do not need an actually _global_ event system because a more local form of communication may suffice. For example, React components often [store shared state in a common ancestor](https://reactjs.org/docs/lifting-state-up.html), passing data _down_ the tree via props or context, and communicating _upwards_ using callbacks or by dispatching actions.
>
> The Liferay event system is useful to communicate between isolated contexts (eg. widgets, iframes).

Reply:
> Based on the context, I think you mean the second one, right?

Yep! 

> I think this whole paragraph might be clearer overall as something like:

Thanks for the wordsmith 💖

Reply:
Let's add those recommendations and push, @diegonvs! The `widgets` addition to the list of "isolated contexts" is particularly interesting because this pattern is more widespread than if it were just to communicate iframes.